diff --git a/js/call_handler.js b/js/call_handler.js
index 9085a3b..498a285 100644
--- a/js/call_handler.js
+++ b/js/call_handler.js
@@ -46,6 +46,15 @@
     }
   }
 
+  function _streamDestroyedHandler(event, callback) {
+    if (typeof callback !== 'function') {
+      callback = function() {};
+    }
+    debug && console.log('DEBUG CallHandler: Stream destroyed ' + event.streams.length);
+
+    callback();
+  }
+
   var CallHandler = {
     /**
      * Setup all the headers needed in the application for any request to the
@@ -75,6 +84,13 @@
           _streamCreatedHandler(event, onStream);
         }
       );
+
+      session.addEventListener(
+        'streamDestroyed',
+        function onStreamDestroyedHandler(event) {
+          _streamDestroyedHandler(event, onStream);
+        }
+      );
     },
     /**
      * Setup all the headers needed in the application for any request to the
